packages/wallets/src/features/cashier/flows/WalletWithdrawal/WalletWithdrawal.tsx

```diff
@@ -1,27 +1,48 @@
 import React, { useEffect, useState } from 'react';
-import { WithdrawalFiatModule, WithdrawalVerificationModule } from '../../modules';
+import { useAuthorize, useCurrencyConfig } from '@deriv/api';
+import { WithdrawalCryptoModule, WithdrawalFiatModule, WithdrawalVerificationModule } from '../../modules';
+import { Loader } from '../../../../components/Loader';
 
 const WalletWithdrawal = () => {
+    const {
+        data: authorizeData,
+        isLoading: isAuthorizeLoading,
+        isSuccess: isAuthorizeSuccess,
+        switchAccount,
+    } = useAuthorize();
+    const { getConfig, isLoading: isCurrencyConfigLoading, isSuccess: isCurrencyConfigSuccess } = useCurrencyConfig();
     const [verificationCode, setVerificationCode] = useState('');
 
     useEffect(() => {
         const queryParams = new URLSearchParams(location.search);
         const verificationQueryParam = queryParams.get('verification');
+        const walletIdParam = queryParams.get('walletid');
 
-        if (verificationQueryParam) {
+        if (verificationQueryParam && walletIdParam) {
             setVerificationCode(verificationQueryParam);
+            switchAccount(walletIdParam); // authorize the intended wallet
 
             const url = new URL(window.location.href);
             url.searchParams.delete('verification'); // Remove the 'verification_code' query parameter
+            url.searchParams.delete('walletid'); // Remove the 'loginid' of the wallet from the query parameters
             window.history.replaceState({}, document.title, url.toString());
         }
     }, []);
 
-    if (verificationCode) {
-        return <WithdrawalFiatModule verificationCode={verificationCode} />;
-    }
-
-    return <WithdrawalVerificationModule />;
+    return (
+        <>
+            {(isAuthorizeLoading || isCurrencyConfigLoading) && <Loader />}
+            {verificationCode && isAuthorizeSuccess && isCurrencyConfigSuccess && authorizeData.currency ? (
+                getConfig(authorizeData.currency)?.type === 'fiat' ? (
+                    <WithdrawalFiatModule verificationCode={verificationCode} />
+                ) : (
+                    <WithdrawalCryptoModule verificationCode={verificationCode} />
```

Comment:
Using `useCurrencyConfig` to check if the _authorized_ wallet is fiat/crypto and rendering the module accordingly.